src/main/java/use_case/EventScheduler.java

```diff
@@ -112,7 +113,7 @@ public boolean checkAvailability(LocalDateTime targetTime, Calendar calendar, Du
                 if (targetTime.isAfter(startTime) && targetTime.isBefore(endTime)) {
                     return false;
                 }
-                else if (targetTime.plus(timeNeeded).isAfter(startTime)) {
+                else if (targetTime.plus(timeNeeded).isAfter(startTime) && targetTime.plus(timeNeeded).isBefore(endTime)) {
```

Comment:
another case is the target time duration covers an existing event

Reply:
fixed